pydis_site/apps/content/resources/guides/python-guides/docker-hosting-guide.md

```diff
@@ -0,0 +1,273 @@
+$ curl -fsSL https://get.docker.com -o get-docker.sh
+$ sudo sh get-docker.sh
+```
+
+## Creating Dockerfile
+
+To tell Docker what it has to do to run the application, we need to create a file named `Dockerfile` in our project's
+root.
+
+1. First we need to specify the *base image*. Doing that will make Docker install some apps we need to run our bot, for
+   example the Python interpreter
+
+```dockerfile
+FROM python:3.10-bullseye
+```
+
+2. Next, we need to copy our requirements to some directory *inside the container*. Let's call it `/app`
+
+```dockerfile
+COPY requirements.txt /app/
+```
+
+3. Now we need to set the directory as working and install the requirements
+
+```dockerfile
+WORKDIR /app
+RUN pip install -r requirements.txt
+```
+
+4. The only thing that is left to do is to copy the rest of project's files and run the main executable
+
+```dockerfile
+COPY . .
+CMD ["python3", "main.py"]
+```
+
+The final version of Dockerfile looks like this:
+
+```dockerfile
+FROM python:3.10-bullseye
+COPY requirements.txt /app/
+WORKDIR /app
+RUN pip install -r requirements.txt
+COPY . .
+CMD ["python3", "main.py"]
+```
+
+## Building Image and Running Container
+
+Now update the project on your VPS and we can run the bot with Docker.
+
+1. Build the image (dot at the end is very important)
+
+```shell
+$ docker build -t mybot .
+```
+
+2. Run the container
+
+```shell
+$ docker run -d --name mybot mybot:latest
+```
+
+3. Read bot logs (keep in mind that this utility only allows to read STDERR)
+
+```shell
+$ docker logs -f mybot
+```
+
+If everything went successfully, your bot will go online and will keep running!
+
+## Using docker-compose
+
+Just 2 commands to run a container is cool but we can shorten it down to just 1 simple command. For that, create
+a `docker-compose.yml` file in project's root and fill it with the following contents:
+
+```yml
+version: "3.8"
+services:
+  main:
+    build: .
+    container_name: mybot
+```
+
+Update the project on VPS, remove the previous container with `docker rm -f mybot` and run this command
+
+```shell
+docker-compose up -d --build
+```
+
+Now the docker will automatically build the image for you and run the container.
+
+## Creating Volumes
+
+The files creating during container run are destroyed after its recreation. To prevent some files from getting
+destroyed, we need to use *volumes* that basically save the files from directory inside of container somewhere on drive.
+
+1. Create a new directory somewhere and copy path to it
+
+```shell
+$ mkdir mybot-data && echo $(pwd)/mybot-data
+```
+
+My path is `/home/exenifix/mybot-data`, yours is most likely different.
+
+2. In your project, store the files that need to be persistant in a separate directory (eg. `data`)
+3. Add the `volumes` construction to `docker-compose` so it looks like this:
+
+```yml
+version: "3.8"
+services:
+  main:
+    build: .
+    container_name: mybot
+    volumes:
+      - /home/exenifix/mybot-data:/app/data
```

Comment:
We wouldn't want people copy pasting this and it not working
```suggestion
      - /home/<username>/mybot-data:/app/data
```

Reply:
There's a pretty explicit statement
https://github.com/Exenifix/site/blob/main/pydis_site/apps/content/resources/guides/python-guides/docker-hosting-guide.md?plain=1#L195